### ISSUE TITLE
Affinity integration

### ISSUE THREAD

**Issue by brianmurray333** [brianmurray333/YammerGroup#1]
**Goal**
Sync company and people lists from Affinity into a web app.

**Steps**
- Create an input modal that follows the below design. The modal allows user to insert and save their Affinity API Key.
- Once saved, sync Company and People List Names from Affinity. Do not sync full lists on this step, just the names of the Lists. 
- In the same modal, present a new field below the API key field titled "Select Lists to Add". The field is a multi-select dropdown menu with type-ahead filtering. It allows users to select specific Lists from all the Lists that were synced over.
- The button on the modal says Done. After the user selects Lists in the step above and clicks the Done button, a file (JSON works) per selected list is prepared (this file will be used in a separate part of the app but for the purposes of this component, the List file preparation is the last step).  

**Notes**
- User needs to be able to edit their API Key and add Lists that they choose to sync. 
- Affinity's API docs: https://api-docs.affinity.co/#introduction
- Modal design: 
![Image](https://github.com/brianmurray333/YammerGroup/assets/3858000/35dcdb0b-3b91-4ecd-af5f-e8e31317cb32)

**Comment by Toheeb-Ojuolape** [comment]
Hi Brian, I'm currently working on this issue. I'm almost done with the UI part of the task, but I'll like to ask if you can give me an Affinity API secret key to test as I require that for the other aspects of the project, unless it is ok for me to just mock the Affinity integration using dummy data for now

**Comment by brianmurray333** [comment]
Thanks Tobi. You can mock using dummy data for now.

**Comment by brianmurray333** [comment]
Hi Tobi - any update?

**Comment by Toheeb-Ojuolape** [comment]
Hi @brianmurray333 . I opened a Pull request for the issue about 2 weeks ago. Here is the link: https://github.com/brianmurray333/YammerGroup/pull/2

You can read through the README.md I added to the webapp folder of the project in my PR to see how to setup the webapp I built

Apologies for the late response, I was expecting a response on the PR I opened and I didn't get notifications of new comments on this issue.

**Comment by brianmurray333** [comment]
Hi Tobi - sorry for the mix up here. The cross in comms between the PR (which I didn't get a notification on), and Sphinx (which I got disconnected so didn't get a msg from you) caused the disruption.

Thank you for completing this job. What is the best way to pay your bounty? Through Sphinx?

**Comment by Toheeb-Ojuolape** [comment]
Yes, I'd prefer to receive it via Sphinx. Thank you for the opportunity @brianmurray333 . Do let me know if there are any new features you'd like me to work on
